src/Components/Footer/Footer.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { FiltersType } from '../../types/filterTypes';
+import { useApiErrorContext, useFormFocusContext, useTodosContext }
+  from '../../hooks/getContextHooks';
+import { deleteTodo } from '../../api/todos';
+import {
+  deleteTodoAction,
+  setIsSpinningAction,
+  removeIsSpinningAction,
+} from '../../Context/actions/actionCreators';
+
+import { getActiveTodos, getCompletedTodos }
+  from '../../helpers/getFilteredTodos';
+
+export const Footer: React.FC = () => {
+  const { setIsFocused } = useFormFocusContext();
+  const {
+    todos,
+    filter,
+    setFilter,
+    dispatch,
+  } = useTodosContext();
+  const { setApiError } = useApiErrorContext();
+
+  const activeTodosNumber = getActiveTodos(todos).length;
+  const completedTodos = getCompletedTodos(todos);
+  const isClearCompletedInvisible = completedTodos.length === 0;
+
+  const handleClearCompletedClick = () => {
+    setIsFocused(false);
+
+    const deletedTodos = completedTodos.map(({ id }) => {
+      const isSpinningAction = setIsSpinningAction(id);
+
+      dispatch(isSpinningAction);
+
+      return deleteTodo(id)
+        .then(() => {
+          const deleteAction = deleteTodoAction(id);
+
+          dispatch(deleteAction);
+        })
+        .catch((error) => {
+          const removeAction = removeIsSpinningAction(id);
+
+          dispatch(removeAction);
+          setApiError(error);
+        })
+        .finally(() => {
+          setIsFocused(true);
+        });
+    });
+
+    return deletedTodos;
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${activeTodosNumber} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        {(Object.entries(FiltersType))
+          .map(([key, value]) => {
+            const url = value === FiltersType.ALL
+              ? ''
+              : value.toLowerCase();
```

Comment:
You can change the `FiltersType` a little and then you won't need to add an extra check:
```js
export enum FiltersType {
  ALL = '',
  ACTIVE = 'Active',
  COMPLETED = 'Completed',
}
...
const url = value.toLowerCase();
```

Reply:
Brilliant!